hack/e2e.sh

```diff
@@ -2,6 +2,81 @@
 
 set -euo pipefail
 
+function run_upstream_vpa_tests() {
+  if $recommendationOnly
+  then
+    echo "recommendationOnly is enabled. Run the recommender e2e tests in upstream ..."
+    pushd ${SCRIPT_ROOT}/e2e
+    GO111MODULE=on go test -mod vendor ./v1/*go -v --test.timeout=60m --args --ginkgo.v=true --ginkgo.focus="\[VPA\] \[recommender\]" --ginkgo.skip="doesn't drop lower/upper after recommender's restart" --report-dir=/workspace/_artifacts --disable-log-dump
+    V1_RESULT=$?
+    popd
+    echo "v1 recommender test result:" ${V1_RESULT}
+
+    if [ $V1_RESULT -gt 0 ]; then
+      echo "Tests failed"
+      exit 1
+    fi
+  else
+    echo "recommendationOnly is disabled. Run the full-vpa e2e tests in upstream ..."
+    pushd ${SCRIPT_ROOT}/e2e
+    GO111MODULE=on go test -mod vendor ./v1/*go -v --test.timeout=60m --args --ginkgo.v=true --ginkgo.focus="\[VPA\] \[full-vpa\]" --report-dir=/workspace/_artifacts --disable-log-dump
+    V1_RESULT=$?
+    popd
+    echo "v1 full-vpa test result:" ${V1_RESULT}
+
+    if [ $V1_RESULT -gt 0 ]; then
+      echo "Tests failed"
+      exit 1
+    fi
+  fi
+}
+
+function await_for_controllers() {
+  local retries=${1:-10}
+  while [ ${retries} -ge 0 ]; do
+    recommenderReplicas=$(kubectl get deployment vpa-recommender-default -n openshift-vertical-pod-autoscaler -oyaml|yq ".status.replicas")
+    if [[ "$recommenderReplicas" == "null" ]]; then
```

Comment:
Also, if you change to the jsonpath version, it will return an empty string instead of `null` if the field doesn't exist, so this entire `if` clause can be replaced with:
```
recommenderReplicas=${recommenderReplicas:=0}
```
Which will cause it to be zero if null or unset.